src/services/auth.ts

```diff
@@ -1,16 +1,36 @@
 "use strict";
 
-const etapiTokenService = require('./etapi_tokens.js');
-const log = require('./log');
-const sqlInit = require('./sql_init.js');
-const utils = require('./utils');
-const passwordEncryptionService = require('./encryption/password_encryption');
-const config = require('./config');
-const passwordService = require('./encryption/password');
+import etapiTokenService = require('./etapi_tokens');
+import log = require('./log');
+import sqlInit = require('./sql_init');
+import utils = require('./utils');
+import passwordEncryptionService = require('./encryption/password_encryption');
+import config = require('./config');
+import passwordService = require('./encryption/password');
 
 const noAuthentication = config.General && config.General.noAuthentication === true;
 
-function checkAuth(req, res, next) {
+// TODO: We are using custom types for request & response because couldn't extract those pesky express types.
```

Comment:
@types/Express provides the Request, Response and NextFunction types. Thus making the Response and Callback types duplicated, and the Request type can be extended like so:

``` ts
interface AppRequest extends Request {
 headers: {
        authorization?: string;
        "trilium-cred"?: string;
    }
    session: {
        loggedIn: boolean;        
    }
}
```

Reply:
I tried submitting a commit with the proposed changed so it would be easier, but I do not have write permissions in the repo :(

Reply:
@Micosilent , this is nice, thank you!